apps/web/lib/checkForEmptyAssignment.ts

```diff
@@ -0,0 +1,30 @@
+import type { EventTypeAssignedUsers, EventTypeHosts } from "~/event-types/views/event-types-single-view";
+
+// This function checks if EventType requires assignment.
+// returns true: if EventType requires assignment but there is no assignment yet done by the user.
+// returns false: for all other scenarios.
+export function checkForEmptyAssignment({
+  assignedUsers,
+  hosts,
+  isManagedEventType,
+  assignAllTeamMembers,
+  isTeamEvent,
+}: {
+  assignedUsers: EventTypeAssignedUsers;
+  hosts: EventTypeHosts;
+  isManagedEventType: boolean;
+  assignAllTeamMembers: boolean;
+  isTeamEvent: boolean;
+}): boolean {
+  if (isTeamEvent && assignAllTeamMembers) {
+    return false;
+  } else if (
+    isTeamEvent &&
```

Comment:
No need to add `isTeamEvent` check here.

Reply:
Updated with more comments.